File: common/exception/src/main/java/org/yechan/BusinessException.java
package org.yechan;

public class BusinessException extends RuntimeException {
    private final Status status;

    public BusinessException(Status status, String message) {
        super(message);
        this.status = status;
    }

    public BusinessException(String message) {
        super(message);
        this.status = Status.INTERNAL_SERVER_ERROR;
    }

    public Status getStatus() {
        return status;
    }

    public int getHttpStatus() {
        return status.toHttpStatus();
    }
}

File: common/exception/src/main/kotlin/org/yechan/BusinessException.kt
package org.yechan

open class BusinessException : RuntimeException {
    val status: Status

    constructor(
        status: Status,
        message: String,
    ) : super(message) {
        this.status = status
    }

    constructor(message: String) : super(message) {
        status = Status.INTERNAL_SERVER_ERROR
    }

    fun getHttpStatus() = status.toHttpStatus()
}
